### PR TITLE
Fix invalid file descriptor

### LINKED ISSUE #632: (cc-oci-runtime:3934): GLib-WARNING **: Invalid file descriptor.
cc-oci-runtime sometimes fails when hooks do not read from stdin
```
"hooks": {
        "prestart": [
        {
            "path": "/bin/echo",
            "args": ["this", "is", "a", "test"],
            "env":  [ "key1=value1"]
        },
        {
            "path": "/bin/ls"
        }
    	],
        "poststart": [
	{
            "path": "/bin/ls",
            "timeout": 5
	}
        ],
        "poststop": [
	{
            "path": "/bin/echo",
            "args": ["other", "test"]
	}
        ]
    },
```

### PR DESCRIPTION
**Pull request by devimc**
This PR simplifies the way we spawn hooks.
Currently hooks that do not read the container
state from stdin, and/or ends before cc-oci-runtime
sends the container state through stdin will fail.

fixes #632

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

File: src/process.c [hunk 1]
```diff
@@ -319,203 +268,147 @@ cc_oci_output_watcher(GIOChannel* channel, GIOCondition cond,
...
+	//parent
+	close_if_set (stdin_pipe[0]);
+	close_if_set (pipe_child_error[1]);
+	close_if_set (pipe_parent_error[0]);
+
+	/* if needed resize pipe size */
+	pipe_sz = fcntl (stdin_pipe[1], F_GETPIPE_SZ);
+	if (pipe_sz < state_length) {
+		if (fcntl (stdin_pipe[1], F_SETPIPE_SZ, state_length+1) < 0) {
+			g_critical ("failed to chage pipe size: %s", strerror(errno));
```
Comment (jodh-intel) [inline src/process.c]: Typo: `change`.

**Comment by jodh-intel** [comment]
This change looks very good. We just need to get a clean test run.

@chavafg - could you re-run the tests to see if the issues are transitory?

File: src/process.c (hunk 1 above)
Comment (devimc) [inline src/process.c]: thanks, fixing it

**Comment by jodh-intel** [comment]
Sounds like we're stuck in a catch-22 situation where the tests won't pass until we land both this PR + 634.

lgtm.

**Comment by chavafg** [comment]
Got some issues regarding #623

**Comment by chavafg** [comment]
qa-passed

**Comment by sameo** [comment]
@devimc Please add a description of the PR.

**Comment by sameo** [comment]
LGTM

**Pull request merged**